mixer/pkg/checkcache/cache.go

```diff
@@ -81,65 +108,20 @@ func New(capacity int32) *Cache {
 		getTime:     time.Now,
 	}
 
-	cc.cacheWrites = prometheus.NewCounterFunc(prometheus.CounterOpts{
-		Namespace: "mixer",
-		Subsystem: "checkcache",
-		Name:      "cache_writes",
-		Help:      "The number of times state in the cache was added or updated.",
-	}, func() float64 {
-		s := cc.cache.Stats()
-		return float64(s.Writes)
-	})
-
-	cc.cacheHits = prometheus.NewCounterFunc(prometheus.CounterOpts{
-		Namespace: "mixer",
-		Subsystem: "checkcache",
-		Name:      "cache_hits",
-		Help:      "The number of times a cache lookup operation succeeded to find an entry in the cache.",
-	}, func() float64 {
-		s := cc.cache.Stats()
-		return float64(s.Hits)
-	})
-
-	cc.cacheMisses = prometheus.NewCounterFunc(prometheus.CounterOpts{
-		Namespace: "mixer",
-		Subsystem: "checkcache",
-		Name:      "cache_misses",
-		Help:      "The number of times a cache lookup operation failed to find an entry in the cache.",
-	}, func() float64 {
-		s := cc.cache.Stats()
-		return float64(s.Misses)
-	})
-
-	cc.cacheEvictions = prometheus.NewCounterFunc(prometheus.CounterOpts{
-		Namespace: "mixer",
-		Subsystem: "checkcache",
-		Name:      "cache_evictions",
-		Help:      "The number of entries that have been evicted from the cache.",
-	}, func() float64 {
-		s := cc.cache.Stats()
-		return float64(s.Evictions)
-	})
-
-	_ = prometheus.Register(cc.cacheWrites)
-	_ = prometheus.Register(cc.cacheHits)
-	_ = prometheus.Register(cc.cacheMisses)
-	_ = prometheus.Register(cc.cacheEvictions)
+	_ = view.Register(writesView, hitsView, missesView, evictionsView)
 
 	return cc
 }
 
 // Close releases any resources used by the check cache.
 func (cc *Cache) Close() error {
-	_ = prometheus.Unregister(cc.cacheWrites)
-	_ = prometheus.Unregister(cc.cacheHits)
-	_ = prometheus.Unregister(cc.cacheMisses)
-	_ = prometheus.Unregister(cc.cacheEvictions)
+	view.Unregister(writesView, hitsView, missesView, evictionsView)
 	return nil
 }
 
 // Get looks up an attribute bag in the cache.
 func (cc *Cache) Get(attrs attribute.Bag) (Value, bool) {
+	defer func() { cc.recordStats() }()
```

Comment:
Here and elsewhere in this file, I'm not thrilled with "defer" here since it induces an allocation (last I looked)

Reply:
Removed.